tests/hypotests/test_calculators.py

```diff
@@ -48,29 +49,71 @@ def test_base_calculator():
+    assert calc_loss.obs_nll([mean_poi[0]]) == mean_nll[0]
+    assert calc_loss.obs_nll([mean_poi[1]]) == mean_nll[1]
+    assert calc_loss.obs_nll([mean_poi[2]]) == mean_nll[2]
 
     mean_poialt = POI(mean, 1.2)
 
-    with pytest.raises(NotImplementedError):
-        calc_loss.pvalue(poinull=mean_poi, poialt=mean_poialt)
-    with pytest.raises(NotImplementedError):
-        calc_loss.expected_pvalue(poinull=mean_poi, poialt=mean_poialt, nsigma=np.arange(-2, 3, 1))
-    with pytest.raises(NotImplementedError):
-        calc_loss.expected_poi(poinull=mean_poi, poialt=mean_poialt, nsigma=np.arange(-2, 3, 1))
+    if calculator == BaseCalculator:
+        with pytest.raises(NotImplementedError):
+            calc_loss.pvalue(poinull=[mean_poi], poialt=[mean_poialt])
+        with pytest.raises(NotImplementedError):
+            calc_loss.expected_pvalue(poinull=[mean_poi], poialt=[mean_poialt], nsigma=np.arange(-2, 3, 1))
+        with pytest.raises(NotImplementedError):
+            calc_loss.expected_poi(poinull=[mean_poi], poialt=[mean_poialt], nsigma=np.arange(-2, 3, 1))
+    else:
+        calc_loss.pvalue(poinull=[mean_poi], poialt=[mean_poialt])
+        with pytest.raises(NotImplementedError):
+            calc_loss.pvalue(poinull=[mean_poi]*2, poialt=[mean_poialt]*2)
+        calc_loss.expected_pvalue(poinull=[mean_poi], poialt=[mean_poialt], nsigma=np.arange(-2, 3, 1))
+        with pytest.raises(NotImplementedError):
+            calc_loss.expected_pvalue(poinull=[mean_poi]*2, poialt=[mean_poialt]*2, nsigma=np.arange(-2, 3, 1))
+        calc_loss.expected_poi(poinull=[mean_poi], poialt=[mean_poialt], nsigma=np.arange(-2, 3, 1))
+        with pytest.raises(NotImplementedError):
+            calc_loss.expected_poi(poinull=[mean_poi]*2, poialt=[mean_poialt]*2, nsigma=np.arange(-2, 3, 1))
 
     model = calc_loss.model[0]
     sampler = model.create_sampler(n=10000)
+    assert is_valid_data(sampler)
 
-    calc_loss.loss_builder(model=[model], data=[sampler], weights=None)
+    loss = calc_loss.lossbuilder(model=[model], data=[sampler], weights=None)
+    assert is_valid_loss(loss)
 
     with pytest.raises(ValueError):
-        calc_loss.loss_builder(model=[model, model], data=[sampler])
+        calc_loss.lossbuilder(model=[model, model], data=[sampler])
     with pytest.raises(ValueError):
-        calc_loss.loss_builder(model=[model], data=[sampler, calc_loss.data])
+        calc_loss.lossbuilder(model=[model], data=[sampler, calc_loss.data[0]])
     with pytest.raises(ValueError):
-        calc_loss.loss_builder(model=[model], data=[sampler], weights=[])
+        calc_loss.lossbuilder(model=[model], data=[sampler], weights=[])
     with pytest.raises(ValueError):
-        calc_loss.loss_builder(model=[model], data=[sampler], weights=[np.ones(10000), np.ones(10000)])
+        calc_loss.lossbuilder(model=[model], data=[sampler], weights=[np.ones(10000), np.ones(10000)])
+
+
+def test_asymptotic_calculator():
```

Comment:
Test should be targeted. I woud separate tests that raise NotImplementedError from other tests that do run implemented stuff. Seems a minimum.